### ISSUE TITLE
RemoveItem on QGraphicsScene

### ISSUE THREAD

**Issue by rashwell** [therecipe/qt#170]
This might be related to #89 in some way.  My Application is way to big at this point to share its issue so I have constructed a tiny example here based on a prior example crafted in Issue #89 

Basically RemoveItem on a QGraphicsScene object works well for me removing just about any QGraphicsItem, or QGraphicsPixmapItem etc.  But as you should see in the below simple example using RemoveItem on a QGraphicsTextItem doesn't work as expected for me.  

```go
package main

import (
	"os"

	"github.com/therecipe/qt/core"
	"github.com/therecipe/qt/gui"
	"github.com/therecipe/qt/widgets"
)

func main() {
	widgets.NewQApplication(len(os.Args), os.Args)

	var (
		view  = widgets.NewQGraphicsView(nil)
		scene = widgets.NewQGraphicsScene3(0, 0, 300, 300, nil)
		font  = gui.NewQFont2("Arial", 20, 2, false)
	)
	view.SetScene(scene)

	var test = scene.AddText("I should be gone", font)
	test.SetPos(core.NewQPointF3(10, 10))

	var test2 = scene.AddText("I Should be here", font)
	test2.SetPos2(10, 40)

        // This should remove the first QGraphicsTextItem from the scene
	scene.RemoveItem(test)

	view.Show()

	widgets.QApplication_Exec()
}
```

**Comment by therecipe** [comment]
Hey

I think you may be lucky, because the example you shared already works for me :)

It was probably fixed with https://github.com/therecipe/qt/commit/11aa6b19de5c193b9b4b5eb1c1707cae3c4f25da to solve this related issue https://github.com/therecipe/qt/issues/156

And I just looked into the code and `scene.RemoveItem` was one of the affected functions.

**Comment by rashwell** [comment]
To follow up, I can confirm you are correct.  I did wipe / fetch / and rebuild with the latest and it works as expected.  Thank you very much, I'll try and stay on top of the latest more often, prior to an issue post.  

Oh FYI on the fresh pull even though everything worked I do have a new warning that I don't understand yet, when building the above example.  Again harmless but curious.

time="2017-01-02T18:32:28-06:00" level=warning msg="failed to find moc structs" func=appMoc.generate

**Comment by therecipe** [comment]
Okay, great :)

And yes, I made `qtdeploy` and probably some of the other tools a bit more verbose.
So there can be new warnings/errors now, that were hidden previously.
But I will make the tools quite again in the future.

About this particular error, that is because `qtmoc` which is invoked by `qtdeploy` can't find structs similar to this one [widgets/bridge2](https://github.com/therecipe/qt/blob/master/internal/examples/widgets/bridge2/bridge2.go#L14-L19) in your code. This is needed to generate your own subclasses with custom slots/signals (and maybe properties in the future)

**Comment by rashwell** [comment]
Yes, the whole sub-classing with additional properties, would probably be awesome, if I understand correctly. My current code does a lot of hoop jumping connecting stuff together, but so far I am able to accomplish everything.  though I am worried that if I don't dig in deeper, I am just making it harder on myself later.  I end up tracking way to much Qt stuff in my own structs, I think simply because I have to do it backwards, instead of sub classing qt objects and adding properties to them I end up doing it more from the go side a typical example of this makes in my code:

```go
// PieceItem for display
type PieceItem struct {
	bv              *BoardView  
	piece           PieceType
	square          SquareType
	qpix            *gui.QPixmap
	scaledqpix      *gui.QPixmap
	qpixitem        *widgets.QGraphicsPixmapItem
	targethighlight *widgets.QGraphicsPathItem
}

// BoardView Comment
type BoardView struct {
	gview       *widgets.QGraphicsView
	gscene      *widgets.QGraphicsScene
	size        int
	squaresize  int	
	squarelist  map[SquareType]*PieceItem
	sidetomove  ColorType
	board       *BoardType
	game        *Game
	thighlights []*widgets.QGraphicsPathItem
	tsquares    []SquareType
	indicator   *widgets.QGraphicsPathItem
	labels      []*widgets.QGraphicsTextItem
}
```

When I think I would rather do something like this instead based on your bridge qtmoc example above:

```go
//go:generate qtmoc
type PieceItem struct {
       *widgets.QGraphicsPixmapItem  // Now I can treat a *PieceItem just like a *QGraphicsPixmapItem
       piece   PieceType // Extra Properties added to my subclassed version
       ...
}
```

Is that where you are headed?

**Comment by therecipe** [comment]
> Is that where you are headed?

Yes, that was the plan :)
But properties will probably made available with something like this (similar to the signal/slot declaration):

```go
//go:generate qtmoc
type PieceItem struct {
       widgets.QGraphicsPixmapItem  // Now I can treat a *PieceItem just like a *QGraphicsPixmapItem
       _   PieceType `prop:"piece"` // Extra Properties added to my subclassed version
       ...
}
```

And then be made available with getters and setters like this:
`(*PieceItem) Piece() PieceItem` and `(*PieceItem) SetPiece(PieceItem)`

But there will be some limitations until there is full support for generic lists/containers.
So that there is no support for `[]T` or `map[T]T` properties in the beginning.

And the supported properties will be limited to primitives and Qt derived classes.

I will see if I can get it working in the next few days, as there should be only the need to make some changes to `qtmoc` to get this initially working.

Also, if you want to get going with `qtmoc` in general then you need to keep in mind that the class you want to subclass `widgets.QGraphicsPixmapItem` in your example, needs to be used without the `*` prefix in your moc struct. (Like in my snipped above)

**Comment by therecipe** [comment]
@rashwell 

Custom properties are now supported with https://github.com/therecipe/qt/commit/1748f5d34a38a04dde6dbae1ba92c891d08d2fcc

and here is an example that shows how to use them https://github.com/therecipe/qt/tree/master/internal/examples/qml/prop2

There will be getter + setter functions and the *Changed signal created automatically for each property.
Also in the future it might be possible to override/connect the getter + setter functions as well.
